api/request.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"k8s.io/utils/strings/slices"
```

Comment:
We should probably use https://pkg.go.dev/golang.org/x/exp/slices#Contains instead